app/src/main/java/org/smartregister/addo/activity/LoginActivity.java

```diff
@@ -108,28 +108,26 @@ public void setServerURL() {
         try {
             File file = new File(Environment.getExternalStorageDirectory() + File.separator + "Addo", "env_switch.json");
 
-            if (file.exists()) {
+            if (file.exists() && !getSwitchConfigurationsFromFile(file).isNull("env")) {
                 // if the file is there, then  switching has taken place and the data was cleared from the device
                 // Get the environment configurations from the file and set the url based on that
 /*                        Yaml yaml = new Yaml();
                         Map<String, Object> envConfig = (Map<String, Object>) yaml.load(new FileInputStream(file));*/
                 JSONObject envConfig = getSwitchConfigurationsFromFile(file);
```

Comment:
Wouldn't it be better if the `getSwitchConfigurationsFromFile()` call is made once? 
```java IF fileExists() { then getSwitchConfigurationsFromFile(file).isNull("env") as a string then check if that string is null... } ?